src/read.c

```diff
@@ -5361,7 +5361,7 @@ avifResult avifDecoderReset(avifDecoder * decoder)
                 if (!foundItem) {
                     AVIF_CHECKERR(numExtraInputImageItems < AVIF_SAMPLE_TRANSFORM_MAX_NUM_EXTRA_INPUT_IMAGE_ITEMS,
                                   AVIF_RESULT_NOT_IMPLEMENTED);
-                    *category = AVIF_ITEM_SAMPLE_TRANSFORM_INPUT_0_COLOR + numExtraInputImageItems;
+                    *category = (avifItemCategory)(AVIF_ITEM_SAMPLE_TRANSFORM_INPUT_0_COLOR + numExtraInputImageItems);
```

Comment:
@y-guyon Yannis, please see this comment.

Nit: This line and the next line are inconsistent.

In this line, `*category` is of the `avifItemCategory` type, so we need a `avifItemCategory` cast.

In the next line, `alphaCategory` is of the `uint32_t` type (see line 5351), so we don't need a cast.

I wonder if we should declare `alphaCategory` as `avifItemCategory`.

Reply:
I guess I kept the most convenient type for each variable at the time but feel free to align them for consistency.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/2332.